forum/views/users.py

```diff
@@ -0,0 +1,433 @@
+            {"$project": {"username": 1, "course_stats": 1}},
+            {"$unwind": "$course_stats"},
+            {"$match": {"course_stats.course_id": course_id}},
+            {"$sort": sort_criterion},
+            {
+                "$facet": {
+                    "pagination": [{"$count": "total_count"}],
+                    "data": [
+                        {"$skip": (page - 1) * per_page},
+                        {"$limit": per_page},
+                    ],
+                }
+            },
+        ]
+        return pipeline
+
+    def _get_sort_criterion(self, sort_by: str) -> dict[str, Any]:
+        """Get sort criterion based on sort_by parameter."""
+        if sort_by == "flagged":
+            return {
+                "course_stats.active_flags": -1,
+                "course_stats.inactive_flags": -1,
+                "username": -1,
+            }
+        elif sort_by == "recency":
+            return {
+                "course_stats.last_activity_at": -1,
+                "username": -1,
+            }
+        else:
+            return {
+                "course_stats.threads": -1,
+                "course_stats.responses": -1,
+                "course_stats.replies": -1,
+                "username": -1,
+            }
+
+    def _get_paginated_stats(
+        self, course_id: str, page: int, per_page: int, sort_criterion: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Get paginated stats for a course."""
+        pipeline = self._create_pipeline(course_id, page, per_page, sort_criterion)
+        return list(Users().aggregate(pipeline))[0]
+
+    def _get_user_data(
+        self, user_stats: dict[str, Any], exclude_from_stats: list[str]
+    ) -> dict[str, Any]:
+        """Get user data from user stats."""
+        user_data = {"username": user_stats["username"]}
+        for k, v in user_stats["course_stats"].items():
+            if k not in exclude_from_stats:
+                user_data[k] = v
+        return user_data
+
+    def _get_stats_for_usernames(
+        self, course_id: str, usernames: list[str]
+    ) -> list[dict[str, Any]]:
+        """Get stats for specific usernames."""
+        users = Users().get_list()
+        stats_query = []
+        for user in users:
+            if user["username"] not in usernames:
+                continue
+            course_stats = user["course_stats"]
+            if course_stats:
+                for course_stat in course_stats:
+                    if course_stat["course_id"] == course_id:
+                        stats_query.append(
+                            {"username": user["username"], "course_stats": course_stat}
+                        )
+                        break
+        return sorted(stats_query, key=lambda u: usernames.index(u["username"]))
+
+    def get(self, request: Request, course_id: str) -> Response:
+        """Get user course stats."""
+        params = request.GET.dict()
+        page = int(request.GET.get("page", "1"))
```

Comment:
Can we move these default values to constants.py?

Reply:
Done